base_name_search_improved/hooks.py

```diff
@@ -0,0 +1,18 @@
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def uninstall_hook(env):
+    _logger.info("Reverting Patches...")
+    env = api.Environment(env.registry.cursor(), SUPERUSER_ID, {})
```

Comment:
Is there any reason to open a new transaction? You don't close this new cursor, which gives me a warning. Also, there are now remaining smart_search fields for all the models in the base module (maybe because the module is not uninstalled yet in the new registry). I tested using the `env` passed to the method, and it seems to work fine.

Reply:
@StefanRijnhart You are right!